controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -2244,6 +2244,9 @@ boolean compareWriterPositions(Map<Long, Long> position1, Map<Long, Long> positi
         return Futures.loop(() -> from.get() < txnIds.size() && transactionsMap.size() < limit, 
                 () -> getTransactionRecords(epoch, txnIds.subList(from.get(), till.get()), context).thenAccept(txns -> {
             for (int i = 0; i < txns.size(); i++) {
```

Comment:
Instead of the 2 break statements, we could add an additional check to the for loop entry condition `transactionsMap.size<limit`